pkg/api/v1/container.go

```diff
@@ -84,3 +85,26 @@ func (c *ContainerGroup) StopAllWorkspaceContainers(ctx echo.Context) error {
 		"message": "all containers stopped",
 	})
 }
+
+func (c *ContainerGroup) StopContainer(ctx echo.Context) error {
+	workspaceId := ctx.Param("workspaceId")
+	containerId := ctx.Param("containerId")
+
+	state, err := c.containerRepo.GetContainerState(containerId)
+	if err != nil {
+		return HTTPBadRequest("invalid container id")
+	}
+
+	if state.WorkspaceId != workspaceId {
+		return HTTPBadRequest("invalid workspace id")
+	}
+
+	err = c.scheduler.Stop(containerId)
+	if err != nil {
+		return HTTPInternalServerError("failed to stop container")
+	}
+
+	return ctx.JSON(http.StatusOK, map[string]interface{}{
+		"message": "container stopped",
```

Comment:
Just a couple of nitpicks. 

Can we capitalize the messages? e.g. "Invalid container ID"

I think the OK response should just be a NoContent response.

Reply:
updated. also made the get responses capitalized